src/app/features/posts/services/comment.service.ts

```diff
@@ -0,0 +1,30 @@
+import { Injectable } from '@angular/core';
+import { Observable } from 'rxjs';
+import { HttpClient } from '@angular/common/http';
+
+@Injectable({
+  providedIn: 'root'
+})
+export class CommentService {
+  constructor(private readonly http: HttpClient) {}
+
+  add(comment: any): Observable<any> {
+    return this.http.post('/api/comments', comment);
+  }
+
+  getByPostId(
+    postId: number,
+    pageNumber: number = null,
+    pageLimit: number = 10
+  ): Observable<any> {
+    const query =
+      '/api/posts/' +
+      postId +
+      '/comments' +
+      (pageNumber
+        ? '?_page=' + pageNumber + (pageLimit ? '&_limit=' + pageLimit : '')
+        : '');
+
```

Comment:
String interpolation is proffered for complex logic:
```typescript
const query = `/api/posts/${postId}/comments/${pageNumber},${pageLimit}`
```
Also use routes.json to abstract the pagination.
"/posts/:postId/comments/:pageNo,:pageSize": "/posts/:postId/comments?_page=:pageNo&_limit=:pageSize"